cmd/agent.go

```diff
@@ -98,14 +98,8 @@ WAIT:
 		goto WAIT
 	}
 
-	// Check if we should do a graceful leave
-	graceful := false
-	if sig == syscall.SIGTERM || sig == os.Interrupt {
-		graceful = true
-	}
-
 	// Fail fast if not doing a graceful leave
-	if !graceful {
+	if sig != syscall.SIGTERM && sig != os.Interrupt {
```

Comment:
All signals except SIGTERM or interrupt are becoming graceful, why?

Reply:
No, only SIGTERM and interrupt are graceful. For any other signal, the condition evaluates to true, because it is neither SIGTERM nor interrupt and then `return 1` causes non-graceful termination.
The code is equivalent.